app/assets/javascripts/govuk_publishing_components/analytics-ga4/gtm-page-views.js

```diff
@@ -0,0 +1,96 @@
+;(function (global) {
+  'use strict'
+
+  var GOVUK = global.GOVUK || {}
+
+  GOVUK.Gtm = {
+    sendPageView: function () {
+      if (window.dataLayer) {
+        var data = {
+          event: 'config_ready',
+          page: {
+            location: this.getLocation(),
+            referrer: this.getReferrer(),
+            title: this.getTitle(),
+            status_code: this.getStatusCode()
+          },
+          publishing: {
+            document_type: this.getMetaContent('format'),
+            publishing_app: this.getMetaContent('publishing-application'),
+            rendering_app: this.getMetaContent('rendering-application'),
+            schema_name: this.getMetaContent('schema-name'),
+            content_id: this.getMetaContent('content-id')
+          },
+          taxonomy: {
+            browse_topic: this.getMetaContent('section'),
+            main_topic: this.getMetaContent('taxon-slug'),
+            main_topic_id: this.getMetaContent('taxon-id'),
+            level_1: this.getMetaContent('themes'),
+            topics: this.getMetaContent('taxon-slugs'),
+            topics_ids: this.getMetaContent('taxon-ids')
+          },
+          content: {
+            language: this.getLanguage(),
+            history: this.getHistory(),
+            withdrawn: this.getWithDrawn(),
+            government: this.getMetaContent('government'),
+            political_status: this.getMetaContent('political-status'),
+            org_ids: this.getMetaContent('analytics:organisations'),
+            world_location: this.getMetaContent('analytics:world-locations')
+          }
+        }
+        window.dataLayer.push(data)
+      }
+    },
+
+    getLocation: function () {
+      return document.location.href
+    },
+
+    getReferrer: function () {
+      return document.referrer
+    },
+
+    getTitle: function () {
+      return document.title
+    },
+
+    // window.httpStatusCode is set in the source of the error page in static
+    // https://github.com/alphagov/static/blob/main/app/views/root/_error_page.html.erb#L32
+    getStatusCode: function () {
+      if (window.httpStatusCode) {
+        return window.httpStatusCode
+      } else {
+        return 200
+      }
+    },
+
+    getMetaContent: function (name) {
+      var tag = document.querySelector('meta[name="govuk:' + name + '"]')
+      if (tag) {
+        return tag.getAttribute('content').toLowerCase()
+      } else {
+        return 'n/a'
+      }
+    },
+
+    getLanguage: function () {
```

Comment:
Is there a reason this uses `title` instead of the `<html>` tag which would have `<html lang="en">` ? 

(My guess is, if something isn't in English then the `<title>` tag has a `lang` attribute, and if it doesn't have that attribute we assume it's English?)

Some pages don't have a lang tag on their `title ` but do on `html` - for example https://gov.uk or https://www.gov.uk/coronavirus

Reply:
That's a really good point, thanks for picking that up. The example page I was building this on had it on the `title` tag and it hadn't occurred to me that it would be somewhere else. I'll check a few more examples and rework this.

Reply:
I've updated the code to use the lang attribute on the `html` element instead - fairly sure from looking through our code that we use that element more consistently for that purpose.

Reply:
Thanks @andysellick - looks good! I'm guessing the `if (html)` condition doesn't need an `else` condition since the likelihood of `html` not existing is near impossible?

Reply:
That's a good point! Thanks, have amended.